zeusd/src/devices/cpu/mod.rs

```diff
@@ -0,0 +1,170 @@
+// RAPL CPU
+// Real RAPL interface.
+#[cfg(target_os = "linux")]
+mod linux;
+#[cfg(target_os = "linux")]
+pub use linux::RaplCpu;
+
+// Fake Rapl interface for dev and testing on macOS.
+#[cfg(target_os = "macos")]
+mod macos;
+#[cfg(target_os = "macos")]
+pub use macos::RaplCpu;
+
+use serde::Serialize;
+use std::path::PathBuf;
+use std::sync::RwLock;
+use std::time::Instant;
+use tokio::sync::mpsc::{Sender, UnboundedReceiver, UnboundedSender};
+use tracing::Span;
+
+use crate::error::ZeusdError;
+
+pub struct PackageInfo {
+    pub index: u32,
+    pub name: String,
+    pub energy_uj_path: PathBuf,
+    pub max_energy_uj: u64,
+
+    num_wraparounds: RwLock<u64>,
+}
+
+#[derive(Serialize)]
+pub struct RaplResponse {
+    cpu_energy_uj: Option<u64>,
+    dram_energy_uj: Option<u64>,
+}
+
+pub trait CpuManager {
+    fn device_count() -> Result<u32, ZeusdError>;
+    fn get_available_fields(index: u32) -> Result<(PackageInfo, Option<PackageInfo>), ZeusdError>;
+    fn get_cpu_energy(&self) -> Result<u64, ZeusdError>;
+    fn get_dram_energy(&self) -> Result<Option<u64>, ZeusdError>;
+    fn stop_monitoring(&mut self);
+}
```

Comment:
This could really use some comments, defining what each function does. For instance, for the two get-energy functions, they get the **cumulative energy number of each device** after compensating for wraparounds. Stuff like this can be pretty helpful.